seed/building_sync/building_sync.py

```diff
@@ -5,230 +5,139 @@
-                "key_path_name": "auc:FloorAreaType",
-                "key_path_value": "Gross",
-                "value_path_name": "auc:FloorAreaValue",
-                "required": True,
-                "type": "double",
-            },
-            "net_floor_area": {
-                "path": "auc:Sites.auc:Site.auc:Buildings.auc:Building.auc:FloorAreas.auc:FloorArea",
-                "key_path_name": "auc:FloorAreaType",
-                "key_path_value": "Net",
-                "value_path_name": "auc:FloorAreaValue",
-                "required": False,
-                "type": "double",
-            },
-            "footprint_floor_area": {
-                "path": "auc:Sites.auc:Site.auc:Buildings.auc:Building.auc:FloorAreas.auc:FloorArea",
-                "key_path_name": "auc:FloorAreaType",
-                "key_path_value": "Footprint",
-                "value_path_name": "auc:FloorAreaValue",
-                "required": False,
-                "type": "double",
-            },
-        }
+class BuildingSync(object):
+    BUILDINGSYNC_V2_PR1 = '2.0-pr1'
+    BUILDINGSYNC_V2_0 = '2.0'
+    VERSION_MAPPINGS_DICT = {
+        BUILDINGSYNC_V2_PR1: BASE_MAPPING_V2_PR1,
+        BUILDINGSYNC_V2_0: BASE_MAPPING_V2_0,
     }
 
     def __init__(self):
         self.filename = None
         self.data = None
         self.raw_data = {}
-
-    @property
-    def pretty_print(self):
-        """
-        Print the JSON to the screen
-        :return: None
-        """
-        print(json.dumps(self.raw_data, indent=2))
+        self.element_tree = None
+        self.version = None
 
     def import_file(self, filename):
         self.filename = filename
 
-        if os.path.isfile(filename):
-            with open(filename, 'rU') as xmlfile:
-                self.raw_data = xmltodict.parse(
-                    xmlfile.read(),
-                    process_namespaces=True,
-                    namespaces={
-                        'http://buildingsync.net/schemas/bedes-auc/2019': 'auc',
-                        'http://www.w3.org/2001/XMLSchema-instance': 'xsi',
-                    }
-                )
-
-        else:
+        if not os.path.isfile(filename):
             raise Exception("File not found: {}".format(filename))
 
+        # save element tree
+        with open(filename) as f:
+            self.element_tree = etree.parse(f)
+
+        # TODO: once xml translator has been implemented and used to convert
+        # files from 2.0-pr1 to 2.0, REMOVE the default argument so it fails
+        # if no proper version is found
+        self.version = self._parse_version(default=self.BUILDINGSYNC_V2_PR1)
+
+        # if the namespace map is missing the auc prefix, fix the tree to include it
+        if self.element_tree.getroot().nsmap.get('auc') is None:
+            self.fix_namespaces()
+
         return True
 
-    def export(self, property_state, process_struct=ADDRESS_STRUCT):
+    def fix_namespaces(self):
```

Comment:
Any better ideas what to do here? The issue is that a user might inculde the right namespace, but not have the prefix as auc, e.g. `xmlns:other_ns="<correct buildingsync uri>"`. I was only able to dig up [this reference](https://bugs.launchpad.net/lxml/+bug/555602) for changing namespace prefixes, and the only thing I got to work was just recreating the tree.

This solution just copies over the entire tree as best as it knows how. The problem with this is I think comments will get dropped. If that's a problem I can spend more time investigating

Reply:
Might be better answered by @nllong or someone else that understands BuildingSync files better.

Reply:
From my understanding the namespace monikers can be mapped to whatever we choose (e.g., auc, n1, n2, etc). I'm not familiar with the impacts of this though. I would like to keep the comments if possible.